fe/fe-core/src/main/java/com/starrocks/scheduler/MvTaskRunProcessor.java

```diff
@@ -71,24 +71,24 @@ public void processTaskRun(TaskRunContext context) throws Exception {
             Preconditions.checkState(materializedView.getPartitions().size() != 0);
             Map<Long, Map<Long, Long>> baseTableVisibleVersionMap =
                     materializedView.getRefreshScheme().getAsyncRefreshContext().getBaseTableVisibleVersionMap();
+            boolean needRefresh = false;
             for (Long baseTableId : baseTableIds) {
                 OlapTable olapTable = (OlapTable) database.getTable(baseTableId);
                 Map<Long, Long> basePartitionVersions =
                         baseTableVisibleVersionMap.computeIfAbsent(baseTableId, k -> Maps.newHashMap());
                 Collection<Partition> partitions = olapTable.getPartitions();
-                boolean needRefresh = false;
                 for (Partition partition : partitions) {
                     long basePartitionId = partition.getId();
                     long basePartitionVersion = partition.getVisibleVersion();
                     Long oldVersion = basePartitionVersions.get(basePartitionId);
-                    if (oldVersion == null || basePartitionVersion > oldVersion.longValue()) {
+                    if (oldVersion == null || basePartitionVersion > oldVersion) {
```

Comment:
this logic has been modified by https://github.com/StarRocks/starrocks/pull/8041
you can pay attention to it.